src/test/java/rx/SingleTest.java

```diff
@@ -1353,4 +1386,364 @@ public String call() throws Exception {
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestObservable other = new TestObservable(sOther);
+        Throwable error = new Throwable();
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Observable.create(other));
+        stringSingle.subscribe(result);
+        other.sendOnError(error);
+
+        result.assertError(error);
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testTakeUntilOtherCompleted() {
+        Subscription sSource = mock(Subscription.class);
+        Subscription sOther = mock(Subscription.class);
+        TestSingle source = new TestSingle(sSource);
+        TestObservable other = new TestObservable(sOther);
+
+        TestSubscriber<String> result = new TestSubscriber<String>();
+        Single<String> stringSingle = Single.create(source).takeUntil(Observable.create(other));
+        stringSingle.subscribe(result);
+        other.sendOnCompleted();
+
+        result.assertError(CancellationException.class);
+        verify(sSource, times(1)).unsubscribe();
+        verify(sOther, times(1)).unsubscribe();
+
+    }
+
+    private static class TestObservable implements Observable.OnSubscribe<String> {
+
+        Observer<? super String> observer = null;
+        Subscription s;
+
+        public TestObservable(Subscription s) {
+            this.s = s;
+        }
+
+        /* used to simulate subscription */
+        public void sendOnCompleted() {
+            observer.onCompleted();
+        }
+
+        /* used to simulate subscription */
+        public void sendOnNext(String value) {
+            observer.onNext(value);
+        }
+
+        /* used to simulate subscription */
+        public void sendOnError(Throwable e) {
+            observer.onError(e);
+        }
+
+        @Override
+        public void call(Subscriber<? super String> observer) {
+            this.observer = observer;
+            observer.add(s);
+        }
+    }
+
+    private static class TestSingle implements Single.OnSubscribe<String> {
+
+        SingleSubscriber<? super String> subscriber = null;
+        Subscription s;
+
+        public TestSingle(Subscription s) {
+            this.s = s;
+        }
+
+        /* used to simulate subscription */
+        public void sendOnSuccess(String value) {
+            subscriber.onSuccess(value);
+        }
+
+        /* used to simulate subscription */
+        public void sendOnError(Throwable e) {
+            subscriber.onError(e);
+        }
+
+        @Override
+        public void call(SingleSubscriber<? super String> observer) {
+            this.subscriber = observer;
+            observer.add(s);
+        }
+    }
+
+    @Test
+    public void testUntilFires() {
```

Comment:
Well, since you placed it in `SingleTest` it should have `takeUntil` prefix
because `SingleTest `contains a lot of tests for different operators and we
should be able to understand what does each test check from its name.

On Mon, 15 Feb 2016, 14:02 Zac Sweers <notifications@github.com> wrote:

> In src/test/java/rx/SingleTest.java
> <https://github.com/ReactiveX/RxJava/pull/3712#discussion_r52933591>:
>
> > +        }
> > +
> > +        /* used to simulate subscription */
> > +        public void sendOnError(Throwable e) {
> > +            subscriber.onError(e);
> > +        }
> > +
> > +        @Override
> > +        public void call(SingleSubscriber<? super String> observer) {
> > +            this.subscriber = observer;
> > +            observer.add(s);
> > +        }
> > +    }
> > +
> > +    @Test
> > +    public void testUntilFires() {
>
> I don't know if it's a typo, I think it's pointing out that this is just
> firing until the until condition fires. This is another one copied over
> from OperatorTakeUntilTest
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/3712/files#r52933591>.
>
-- 

@artem_zin